lib/rspec/mocks/proxy.rb

```diff
@@ -35,7 +35,7 @@ def initialize(object, order_group, options={})
 
       # @private
       def ensure_can_be_proxied!(object)
-        return unless object.is_a?(Symbol) || object.frozen?
+        return unless object.is_a?(Symbol) || (object.is_a?(String) && object.frozen?)
```

Comment:
Cosmetic: This is getting hard to comprehend. Symbol and a frozen string can't be nil, can they (see a check below)?

Reply:
Yeah agreed. These are really only here for the sake of maintaining the existing tests for these two classes. I had originally removed this method entirely because the handling in method double actually handles these same cases just fine. For example this extra test passes just fine:

```
context "when given a symbol" do
  let(:object) { :object }

  it "raises clear error" do
    expect {
      target.to receive(:foo).and_return(:baz)
    }.to raise_error(ArgumentError, /Cannot proxy frozen objects/)
  end
end
```

Just let me know what the preferred behavior for this method is and I'll do it

Reply:
Understood. Let's keep the change to the minimum. I'd go with removing the redundant `nil?` check, and the rest is fine.